src/commons/BarChartData.js

```diff
@@ -0,0 +1,121 @@
+class BarChartData {
+  /**
+   * Validate behavior of graph according to id indicator
+   *
+   * @param {Integer} idIndicator id indicator to be validated
+   */
+  static validateIndicator(idIndicator) {
+    let validIndicator = false;
+    let indicatorNamePositive = 'No definido';
+    let indicatorNameNegative = 'No definido';
+
+    switch (idIndicator) {
+      case 12:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área protegida dentro del bloque';
+        indicatorNameNegative = 'Superficie del área protegida fuera del bloque';
+        break;
+      case 18:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área de reserva de ley 2da de 1959 dentro del bloque';
+        indicatorNameNegative = 'Superficie del área de reserva de ley 2da de 1959 fuera del bloque';
+        break;
+      case 20:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área del territorio étnico dentro del bloque';
+        indicatorNameNegative = 'Superficie del área del territorio étnico fuera del bloque';
+        break;
+      case 22:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área de la ZRC dentro del bloque';
+        indicatorNameNegative = 'Superficie del área de la ZRC fuera del bloque';
+        break;
+      case 24:
+        validIndicator = true;
+        indicatorNamePositive = 'Superficie del área del ecosistema estrategico dentro del bloque';
+        indicatorNameNegative = 'Superficie del área del ecosistema estrategico fuera del bloque';
+        break;
+      default:
+        validIndicator = false;
+        break;
+    }
+
+    return { validIndicator, indicatorNamePositive, indicatorNameNegative };
+  }
+
+  /**
+   * Construct an array to be inserted in the transformed graph data
+   *
+   * @param {Array} rawData
+   * @param {String} valueDescription
+   */
+  static processOneRow(rawData, valueDescription) {
+    const row = {
+      description: valueDescription,
+      value: 0,
+      perc: 0,
+    };
+    const rowComputed = [];
+    rawData.forEach((item) => {
+      if (item.value_description === valueDescription) {
+        if (BarChartData.validateValuePercentage(item.id_indicator)) {
+          row.value = parseFloat(item.indicator_value, 10);
```

Comment:
Please add numberWithCommas(Number(parseFloat(item.indicator_value, 10)).toFixed(2)) or numberWithCommas(Number(row.value).toFixed(2)) to guarantee the number format to 2 decimals


// Thousands number format
const numberWithCommas = (x) => x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');